### ISSUE TITLE
Metricbeat kafka module - mapping conflict

### ISSUE THREAD

**Issue by hartfordfive** [elastic/beats#4667]
**Metricbeat Version: 5.4.3**
**Logstash Version: 5.4.3**
**Operating System: Ubuntu 16.04**

When enabling the Kafka module for metric beat, resulting data is created under the top level `kafka.*` object.  This is ends up causing a mapping conflict if you also have the `decorate_events` parameter set to true as this also adds a top level `kafka` property.  In my specific use case, i have a conflict when it comes to the `kafka.partiton` property.  

In the case of Metricbeat, `kafka.partition` contains an object such as the following:
```
{
  "beat": {
    "name": "kafka01",
    "version": "5.4.3",
    "hostname": "kafka01"
  },
  "kafka": {
    "partition": {
      "topic": {
        "name": "logstash-data"
      },
      "broker": {
        "id": 7,
        "address": "localhost:9092"
      },
      "offset": {
        "oldest": 0,
        "newest": 241
      },
      "partition": {
        "insync_replica": true,
        "replica": 7,
        "leader": 7,
        "id": 29
      }
    }
  },
  "tags": [
    "metricbeat"
  ],
  "@timestamp": "2017-07-13T16:28:42.718Z",
  "metricset": {
    "host": "localhost:9092",
    "module": "kafka",
    "rtt": 649996,
    "name": "partition"
  },
  "type": "metricbeat"
}
``` 

Whereas Logstash adds `kafka.partition` with a concrete string/int value:
```
{
  "kafka": {
      "consumer_group": "cg-logstash-data",
      "partition": 11,
      "offset": 376132,
      "topic": "logstash-data",
      "key": null
    }
}
```

I know this involves two different applications and two different development teams, although it seems like there could be multiple options to avoid this:

### In the Metricbeat Kafka module:
1. Create a configuration that allows you to customize the property in which the kafka properties will be stored.  
2. Simply rename the top level key from `kafka` to `kafka_metrics` in the metricbeat kafka module.

### In the Logstash Kafka input plugin
1. Add a property in the logstash Kafka input plugin allowing you to change the default top level field name for the kafka stats when the `decorate_events => true` parameter is set.
2. Change the default name of the top level property that stores the kafka stats to something such as `kafka_metrics`

Once again, I realize those last two suggestions are beyond the direct scope of the beats library as they relate to another development team, although I'm sure you guys could communicate this with them instead of having two separate but related issues.

**Comment by tsg** [comment]
Thanks @hartfordfive, this is a good find! Pinging @suyograo for awareness.

**Comment by suyograo** [comment]
Thanks for pinging me. I fixed it in https://github.com/logstash-plugins/logstash-input-kafka/pull/213.

**Comment by suyograo** [comment]
Fixed in the Kafka input and available as v7.0.0 of the plugin, so I'm closing this. /cc @tsg
